### PR TITLE
Drupal8 enhancements

### PR DESCRIPTION
**Pull request by rmiroballi**
Advanced Settings:
Users can be matched by email (default) or user name.
Drupal Roles can be mapped to a claim by value. Drupal Roles can be removed from the user if the claim no longer has the role.
CSS for Fallback Login can added.
Drupal profile fields can be mapped to claims.

### CODE REVIEW

File: src/Controller/AuthController.php [hunk 1]
```diff
@@ -438,7 +431,7 @@ protected function createDrupalUser($userInfo) {
     }
 
     // If the username already exists, create a new random one.
-    $username = $userInfo['nickname'];
+    $username = $userInfo['preferred_username'];
```
Comment (mostekcm) [inline src/Controller/AuthController.php]: is this okay to switch?  Any backwards compatibility issue potential here?

**Comment by mostekcm** [comment]
Looks like mostly log changes here.  Plus the username check.  I did not have time to actually test this, just did the code review.  So may want to hold off on merging until someone can test it.

File: src/Controller/AuthController.php (hunk 1 above)
Comment (rmiroballi) [inline src/Controller/AuthController.php]: I will go back and check on this.

**Comment by mostekcm** [comment]
@rmiroballi I merged to drupal 8, the trick is now going to be to merge to master.  There is quite a bit here and it may conflict with previous versions of master, so it may be tricky.  If you are able to test after I merge before pushing to master and creating a new version on drupal, please let me know.

**Pull request merged**